openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -171,17 +172,22 @@ def from_fault_data(cls, fault_trace, upper_seismogenic_depth,
 
     @classmethod
     def get_fault_vertices_3d(cls, fault_trace, upper_seismogenic_depth,
-                              lower_seismogenic_depth, dip):
+                              lower_seismogenic_depth, dip, index_patch=None):
         """
-        Get surface main vertexes.
-
+        Get surface main vertices.
         Parameters are the same as for :meth:`from_fault_data`, excluding
         mesh spacing.
 
+        :param index_patch:
+            indicate the patch of the fault to output the vertices.
+            The first patch strating from where the fault trace starts.
         :returns:
```

Comment:
Marco, I agree with you that the format of the output should be the same in a method. We actually had discussion on it before but at that time we decide to leave like it for the time being. Yes, It was for maintaining the backward compatibility. 

However, I check the code again, the case when the index_patch=None, is no longer used in the get_dpp method, and no where else. The method actual using in the get_dpp method now is only the one specifying the target fault patch. 

Maybe I can just delete the output, which returns the coordinates of the points delimiting the border of the fault surface(the case when index_patch= None). Make the method only return the 4 points of the target fault patch.

If so, for the default index_patch, I will then set up it as 1(index_patch=1), the first patch of the fault.

How do you think?